app/src/main/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreference.kt

```diff
@@ -26,6 +25,6 @@ class RadioSearchEngineListPreference @JvmOverloads constructor(
 
     override fun onSearchEngineSelected(searchEngine: SearchEngine) {
         context.components.search.searchEngineManager.defaultSearchEngine = searchEngine
-        context.settings().defaultSearchEngineName = searchEngine.name
+        context.components.search.provider.setDefaultEngine(context, searchEngine.identifier)
```

Comment:
We should not be accessing this setting directly. It should all go through the `FenixSearchEngineProvider`